### PR TITLE
Add. Accession number for lab test performed in OpenElis to orders table

### PR DESCRIPTION
**Pull request by riyaTw**
Card link - https://bahmni.atlassian.net/browse/BAH-3416?atlOrigin=eyJpIjoiMDdkY2ViNTg1ZTY1NDlhOWIwOTk2ZDEwYWEwZDI1ZjAiLCJwIjoiaiJ9

### CODE REVIEW

File: openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java [hunk 1]
```diff
@@ -250,6 +250,23 @@ private Obs createObsWith(String textValue, Concept concept, Date obsDateTime) {
 
     protected Set<Encounter> associateTestResultsToOrder(OpenElisAccession openElisAccession) throws ParseException {
         Encounter orderEncounter = encounterService.getEncounterByUuid(openElisAccession.getAccessionUuid());
+
+        if (!orderEncounter.getOrders().isEmpty()) {
+            for (OpenElisTestDetail testDetail : openElisAccession.getTestDetails()) {
+                if (testDetail == null && testDetail.getStatus() == null) {
```
Comment (angshu) [inline openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java:256]: should it not be 
> if (testDetail == null || testDetail.getStatus() == null) { continue; }

File: openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java [hunk 2]
```diff
@@ -250,6 +250,23 @@ private Obs createObsWith(String textValue, Concept concept, Date obsDateTime) {
...
+                    continue;
+                }
+                for (Order order : orderEncounter.getOrders()) {
+                    if (testDetail.getTestUuid().equals(order.getConcept().getUuid())) {
+                        if ("Canceled".equals(testDetail.getStatus())) {
```
Comment (angshu) [inline openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java:261]: I hope the spelling is right.

File: openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java (hunk 1 above)
Comment (riyaTw) [reply openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java]: Done

File: openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java (hunk 2 above)
Comment (riyaTw) [reply openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java:261]: Yes its correct

**Review by angshu — APPROVED** [review]

File: openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java [hunk 3]
```diff
@@ -250,6 +250,23 @@ private Obs createObsWith(String textValue, Concept concept, Date obsDateTime) {
...
+            for (OpenElisTestDetail testDetail : openElisAccession.getTestDetails()) {
+                if (testDetail == null || testDetail.getStatus() == null) {
+                    continue;
+                }
+                for (Order order : orderEncounter.getOrders()) {
+                    if (testDetail.getTestUuid().equals(order.getConcept().getUuid())) {
+                        if ("Canceled".equals(testDetail.getStatus())) {
+                            continue;
+                        }
+                        order.setAccessionNumber(openElisAccession.getAccessionUuid());
```
Comment (gokultw) [inline openmrs-elis-atomfeed-client-omod/src/main/java/org/bahmni/module/elisatomfeedclient/api/worker/OpenElisAccessionEventWorker.java:264]: Why we are setting the access uuid? we have the column name as access number.

**Pull request merged**